audio-search/app.py

```diff
@@ -17,14 +20,34 @@
 def config():
     os.environ['JINA_SHARDS'] = str(4)
     os.environ['JINA_WORKSPACE'] = './workspace'
+    os.environ['JINA_DATA_FILE'] = 'data/*.wav'
     os.environ['JINA_PORT'] = os.environ.get('JINA_PORT', str(65481))
     os.environ['JINA_TOPK'] = str(JINA_TOPK)
 
 
 def call_api(url, payload=None, headers={'Content-Type': 'application/json'}):
-    import requests
     return requests.post(url, data=json.dumps(payload), headers=headers).json()
 
+def index_restful(num_docs):
+    f = Flow().load_config('flows/index.yml')
+
+    with f:
+        data_path = os.path.join(os.path.dirname(__file__), os.environ.get('JINA_DATA_FILE', None))
+        print(f'Indexing {data_path}')
+        url = f'http://0.0.0.0:{f.port_expose}/index'
+
+        input_docs = _input_lines(
+            filepath=data_path,
+            size=num_docs,
+            read_mode='r',
+        )
+        data_json = {'data': [Document(text=text).dict() for text in input_docs]}
+        print(f'#### {len(data_json["data"])}')
+        with TimeContext(f'QPS: indexing {num_docs}', logger=f.logger):
```

Comment:
I could happen that `num_docs` is larger than the actual number of available docs. For example, when the user enters a very high number. We could take the min here to make sure that the QPS print is correct.

```
min(num_docs, len(data['data'])
```

Reply:
this is an edge case, as the user knows the dataset for indexing before hand, like say `flickr30k`, user shouldn't put `num_docs` as `40000`. 
however, this could be considered for refactoring all examples for consistency.